core/dnsserver/server.go

```diff
@@ -288,9 +288,21 @@ func (s *Server) OnStartupComplete() {
 	}
 
 	for zone := range s.zones {
+		// split addr into protocol, IP and Port
+		_, ip, port, err := SplitProtocolHostPort(s.Addr)
+
+		if err != nil {
+			// this should not happen, but we need to take care of it anyway
+			fmt.Println(zone + ":" + s.Addr)
+			return
+		}
+		if ip == "" {
+			fmt.Println(zone + ":" + port)
+			return
+		}
 		// if the server is listening on a specific address let's make it visible in the log,
 		// so one can differentiate between all active listeners
-		fmt.Println(zone + " on " + s.Addr)
+		fmt.Println(zone + ":" + port + " on " + ip)
```

Comment:
Changed the printed format so it present the same line as before, when BIND is not involved.
Sample of new log is:

```
com.:6003
.:5001 on 127.0.0.1
.:5002 on 127.0.0.1
.:5002 on ::1
.:5002 on 192.168.1.163

```
for the corefile below:
```
com.:6003 {
    log . "6001 com ::1 {type} {name} - to 5003"
    errors
    proxy . 127.0.0.1:5001 {
        protocol dns force_tcp
    }
}


.:5001 {
    bind 127.0.0.1
    log . "5001 (type} {name}"
    proxy . 8.8.8.8:53 {
        protocol dns force_tcp
    }
}

.:5002 {
    bind 127.0.0.1 ::1 192.168.1.163
    log . "5002 (type} {name}"
    proxy . 8.8.8.8:53 {
        protocol dns force_tcp
    }
}
```